examples/read_module_protocol.cf

```diff
@@ -0,0 +1,44 @@
+bundle agent cache_maintenance
+# Creates a module protocol cache, refreshes it if minute is 30-35
+{
+    vars:
+      "file"
+        string => "$(this.promise_dirname)/cached_module";
+
+  classes:
+    "cache_refresh"
+      if => not(fileexists("$(file)"));
+    Min30_35::
+      "cache_refresh";
+
+  files:
+    cache_refresh::
+      "$(file)"
+        create => "true",
+        edit_template_string => "=my_variable=$(sys.date)",
+        template_data => "{}",
+        template_method => "inline_mustache";
+}
+
+bundle agent demo
+# Demonstrates read_module_protocol function, prints a variable from it
+{
+    classes:
+      "cache_was_read"
+        if => read_module_protocol("$(cache_maintenance.file)");
```

Comment:
Should not have to quote $(cache_maintenance.file) here.

Reply:
I disagree - my style more clearly shows that it expands to a string.